### PR TITLE
feat: Add PermissionSplitter test script scaffolding

### PR DESCRIPTION
**Pull request by nicholaspai**
This  PR was created by running `forge install foundry-rs/forge-std` which created a local `lib/` directory to store 
useful forge contracts for testing, like `forge-std`. I also created a new directory `test/foundry` to store foundry tests from here on out.

I've been running the following test script with `forge test --fork-url https://mainnet.infura.io/v3/<KEY> -vvvv` (the four "v" flags are for for maximum logging)

Fixes ACX-1827

### CODE REVIEW

**Review by mrice32 — APPROVED** [review]
LGTM!

File: test/foundry/PermissionSplitter.t.sol [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        // Calling fake function as admin with value also succeeds and wraps the msg.value
+        // and then does nothing.
+        vm.deal(defaultAdmin, 1 ether);
+        vm.prank(defaultAdmin);
+        (bool success2, bytes memory reason) = address(hubPoolProxy).call{ value: 1 ether }("doesNotExist()");
+        assertTrue(success2);
+        assertEq(address(hubPool).balance, balBefore);
+    }
+
+    /// forge-config: default.fuzz.runs = 300
```
[the remark this answers is not in the document]
Comment (nicholaspai) [reply test/foundry/PermissionSplitter.t.sol:180]: oh ok, so we really only need to compare the Proxy functions with the HubPool functions then?

File: test/foundry/PermissionSplitter.t.sol [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+        // Calling HubPool setPaused directly should fail, even if called by previous owner.
+        vm.startPrank(defaultAdmin);
+        vm.expectRevert(OWNABLE_NOT_OWNER_ERROR);
+        hubPool.setPaused(true);
+        vm.stopPrank();
+
+        // Must call HubPool via PermissionSplitterProxy.
+        vm.prank(pauseAdmin);
+        hubPoolProxy.setPaused(true);
+        assertTrue(hubPool.paused());
```
[the remark this answers is not in the document]
Comment (nicholaspai) [reply test/foundry/PermissionSplitter.t.sol:137]: 887295b8b7a54d53948cca80f70ace18d22fbfb7

File: test/foundry/PermissionSplitter.t.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (mrice32) [reply test/foundry/PermissionSplitter.t.sol]: Yep!

[the remark this answers is not in the document]
Comment (mrice32) [reply test/foundry/PermissionSplitter.t.sol]: But collisions are a superset of that case, so collisions are fine to test for -- just wanted to mention it.

File: test/foundry/PermissionSplitter.t.sol [hunk 3]
```diff
@@ -162,25 +180,62 @@ contract PermissionSplitterTest is Test {
...
+    function testReceive() public {
+        // Sending ETH to the proxy should trigger the target's receive() function, which on the HubPool wraps
+        // any msg.value into wrapped native token.
+        uint256 balBefore = address(hubPool).balance;
+
+        vm.deal(defaultAdmin, 1 ether);
+        vm.prank(defaultAdmin);
+        (bool success, ) = address(hubPoolProxy).call{ value: 1 ether }("");
+        assertTrue(success);
+        assertEq(address(hubPool).balance, balBefore);
```
Comment (mrice32) [inline test/foundry/PermissionSplitter.t.sol:236]: nit: should we also check the balance of the proxy to make sure the ETH got through?

File: test/foundry/PermissionSplitter.t.sol [hunk 4]
```diff
@@ -162,25 +180,62 @@ contract PermissionSplitterTest is Test {
         vm.stopPrank();
     }
 
-    function testFallback() public {
+    function testTransferOwnership() public {
```
Comment (mrice32) [inline test/foundry/PermissionSplitter.t.sol:183]: Nice

File: test/foundry/PermissionSplitter.t.sol [hunk 5]
```diff
@@ -140,6 +146,18 @@ contract PermissionSplitterTest is Test {
         vm.prank(defaultAdmin);
         hubPoolProxy.setPaused(false);
         assertFalse(hubPool.paused());
+
+        // Multiple EOA's can be granted the pause role.
```
Comment (mrice32) [inline test/foundry/PermissionSplitter.t.sol:150]: Ahhh, good call

**Review by mrice32 — APPROVED** [review]
LGTM!

**Review by mrice32 — APPROVED** [review]

**Pull request merged**